pkg/config/settings/runtime_setting_mutex_profile_fraction.go

```diff
@@ -0,0 +1,42 @@
+package settings
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/profiling"
+)
+
+// RuntimeMutexProfileFraction wraps runtime.SetMutexProfileFraction setting.
+type RuntimeMutexProfileFraction (string)
+
+// Name returns the name of the runtime setting
+func (r RuntimeMutexProfileFraction) Name() string {
+	return string(r)
+}
+
+// Description returns the runtime setting's description
+func (r RuntimeMutexProfileFraction) Description() string {
+	return "This setting controls the fraction of mutex contention events that are reported in the mutex profile"
+}
+
+// Hidden returns whether or not this setting is hidden from the list of runtime settings
+func (r RuntimeMutexProfileFraction) Hidden() bool {
+	return false
+}
+
+// Get returns the current value of the runtime setting
+func (r RuntimeMutexProfileFraction) Get() (interface{}, error) {
+	return profiling.GetMutexProfileFraction(), nil
+}
+
+// Set changes the value of the runtime setting
+func (r RuntimeMutexProfileFraction) Set(value interface{}) error {
+	rate, err := GetInt(value)
+	if err != nil {
+		return err
+	}
+
+	profiling.SetMutexProfileFraction(rate)
+	config.Datadog.Set("internal_profiling.mutex_profile_fraction", rate)
```

Comment:
What is the reason to change the config?

Reply:
So that the current values are reflected in the runtime config dump returned by `agent config`